plugins/UserCountry/API.php

```diff
@@ -75,6 +82,14 @@ public function getRegion($idSite, $period, $date, $segment = false)
         $separator = Archiver::LOCATION_SEPARATOR;
         $unk = Visit::UNKNOWN_CODE;
 
+        // show visits tracked as Tibet as region of China
+        $dataTable->filter('GroupBy', array('label', function($label) {
```

Comment:
As those `GroupBy` filters are quite slow I wonder if we could maybe remove this again in Piwik 4 and somehow mention this in an issue or so? Or maybe we could have first a simple check whether there is actually any row containing this label? Like `if($dataTable->getRowFromLabel('1|ti')) { ... groupby filter...} `  

Same could be done in countries. This would not check lowercase but don't know if we actually have it in different cases in the DB?

Reply:
Checking if an entry exists before adding the group by filters should work, will test that.
Removing in Piwik 4 might not be that easy. We could add an update script to update all entries in log_visit table (which I would do in next major release only as it might run a bit longer), so we can remove some of the checks for visitor log & profile, but the group by would still be required if the archived reports aren't reprocessed.

Reply:
It is fine to leave it in there as long as we do the check if table has the row at all.